RubberduckTests/CodeExplorer/CodeExplorerViewModelTests.cs

```diff
@@ -245,6 +245,31 @@ public bool AddUserDocument_CanExecuteBasedOnProjectType(ProjectType projectType
             }
         }
 
+        [Category("Code Explorer")]
+        [Test]
+        [TestCase(ComponentType.ActiveXDesigner, ExpectedResult = true)]
+        [TestCase(ComponentType.ClassModule, ExpectedResult = true)]
+        [TestCase(ComponentType.ComComponent, ExpectedResult = true)]
+        [TestCase(ComponentType.DocObject, ExpectedResult = true)]
+        [TestCase(ComponentType.Document, ExpectedResult = true)]
+        [TestCase(ComponentType.MDIForm, ExpectedResult = true)]
+        [TestCase(ComponentType.PropPage, ExpectedResult = true)]
+        [TestCase(ComponentType.RelatedDocument, ExpectedResult = false, Ignore = "Module doesn't contain members")]
+        [TestCase(ComponentType.ResFile, ExpectedResult = false, Ignore = "Module doesn't contain members")]
```

Comment:
IDK if it's common to comment on my own commit... 😁

These test are ignored since they throw an exception. I can't tell how will that affect the actual CE in IDE. Will it prevent execution? Will it crash?

Reply:
Make sure `Rubberduck.Deployment` is the solution's "StartUp Project" (should be bold): 

![image](https://user-images.githubusercontent.com/5751684/66531996-c972eb80-eadb-11e9-9c4e-df61c00c1085.png)

You can right-click the `Rubberduck.Deployment` project, select "Properties", and in the "Debug" tab you want "Launch" set to "Executable" and "Executable" set to "C:\Program Files (x86)\Microsoft Office\root\Office16\EXCEL.EXE" to build & launch Excel when you hit F5 in Visual Studio 😃

Reply:
Right... But how do I add Related Document or Res File? Aren't they avilable only for VB6 IDE?
These are the two test cases I lack.

Reply:
What is the exception? Is it handled, can/should it be caught & logged? We wouldn't want to ignore tests if they are actually telling us that there's a problem somewhere...

Reply:
The exception is when the `MockedCodeExplorer` is trying to select the first member of `RelatedDocument` or `ResFile` component types, in order to perform the test:
```
using (var explorer = new MockedCodeExplorer(ProjectType.HostProject, componentType, @"Public Sub Foo():  MsgBox """":End Sub ")
                .ImplementExtractIntercaceCommand().SelectFirstModule())
```
Which in my mind could mean one of two things:
1. RD's support for `RelatedDocument` and `ResFile` is not full. (at least in the CE, they don't appear)
2. It is in fact intentionally impossible for the user to select `RelatedDocument` or `ResFile` in CE.

In both cases, the exception has nothing to do with the test itself, but with the setup. i.e. the test fails because it's unable to select the first module.
I think we can safely assume that the tests are fine, since the only thing we really are testing here is `.Where(item => ModuleTypes.Contains(item.DeclarationType))` provided with
```
private static readonly DeclarationType[] ModuleTypes =
        {
            DeclarationType.ClassModule,
            DeclarationType.Document,
            DeclarationType.UserForm
        };
```
so `RelatedDocument` and `ResFile` are excluded...

Determining how to proceed with them in the CE is a matter for a different PR.

Reply:
Pinging @mansellan for opinions on the VB6 stuff above.